src/cli/net/mod.rs

```diff
@@ -1,12 +1,78 @@
-pub use data::Command;
+// need this to allow MessageEnvelope to have custom serde bounds
+#![allow(clippy::type_repetition_in_bounds)]
 
 mod client;
 mod data;
 mod server;
 
-pub use server::{bind as bind_mpc_helper_server, router as mpc_helper_router, BindTarget};
-
+pub use client::{MpcHandle, MpcHttpConnection as Client};
+pub use data::Command;
 #[cfg(feature = "self-signed-certs")]
 pub use server::tls_config_from_self_signed_cert;
+pub use server::{bind as bind_mpc_helper_server, router as mpc_helper_router, BindTarget};
+use std::str::FromStr;
 
-pub use client::{MpcHandle, MpcHttpConnection as Client};
+use crate::cli::net::server::MpcServerError;
+use crate::protocol::{QueryId, RecordId};
+use async_trait::async_trait;
+use axum::body::Bytes;
+use axum::extract::{FromRequest, RequestParts};
+use axum::http::header::HeaderName;
+
+static OFFSET_HEADER_NAME: HeaderName = HeaderName::from_static("offset");
+static DATA_SIZE_HEADER_NAME: HeaderName = HeaderName::from_static("data-size");
+
+pub struct RecordHeaders {
+    offset: usize,
+    data_size: usize,
```

Comment:
would it be ever different from [`content-length`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Length) header?

Reply:
i thought `content-length` refers to the size of the entire message? this should be the length of a single piece of data within the message

Reply:
leave docs to clarify

Reply:
make a GH issue to alias offset type, so we can change max size of data at later date

Reply:
https://github.com/private-attribution/ipa/issues/166